website/public/js/controllers/groupsCtrl.js

```diff
@@ -16,6 +16,13 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Shared', 'Groups', '
     }
 
     $scope.isMemberOfGroup = function(userid, group){
+
+      // If the group is a guild, just check for an intersection with the
+      // current user's guilds, rather than checking the members of the group.
+      if(group.type === 'guild') {
+        return _.detect(Groups.myGuilds(), function(g) { return g._id === group._id });
```

Comment:
Yep, that was it. When you're on the public guilds page, and you click the join button, it takes you to the guild, but takes a couple seconds to refresh. So for about 2-3 seconds I panicked because it was still showing the join button. 

I've removed the ternary and adjusted the tests to check `to.be.ok;` and `to.not.be.ok;`